lib/usdUfe/utils/usdUtils.cpp

```diff
@@ -261,6 +371,53 @@ bool updateReferencedPath(const UsdPrim& oldPrim, const SdfPath& newPath)
     return true;
 }
 
+bool cleanReferencedPath(const UsdPrim& deletedPrim)
+{
+    SdfChangeBlock changeBlock;
+
+    for (const auto& p : deletedPrim.GetStage()->Traverse()) {
+
+        auto primSpec = getPrimSpecAtEditTarget(p);
+        // check different composition arcs
+        if (p.HasAuthoredReferences()) {
+            if (primSpec) {
+
+                SdfReferencesProxy referencesList = primSpec->GetReferenceList();
+
+                // update append/prepend lists individually
+                removeInternalReferencePath(deletedPrim, referencesList, SdfListOpTypeAppended);
```

Comment:
Even in the previously-existing version for rename, IDK why we don't treat the SdfListOpTypeOrdered, SdfListOpTypeAdded, SdfListOpTypeExplicit and SdfListOpTypeDeleted. MayaUSd might not author those, but we can edit USD file that may contain them.